ts_src/bufferutils.ts

```diff
@@ -87,7 +98,7 @@ export class BufferWriter {
     this.offset += slice.copy(this.buffer, this.offset);
   }
 
-  writeVarSlice(slice: Buffer): void {
+  writeVarSlice(slice: Buffer = Buffer.alloc(1)): void {
```

Comment:
add a comment for this? can be unexpected if used outside

Reply:
Finally, I've removed default value and handle undefined/null at upper level